### ISSUE TITLE
Document recommended method(s) to set environment variables when environments containing a package are activated

### ISSUE THREAD

**Issue by mfisher87** [conda/conda-build#5035]
I help maintain a Conda Forge feedstock, and a user of the package [reported problems using the package with Fish](https://github.com/conda-forge/quarto-feedstock/issues/30). The root cause was our [activation scripts](https://docs.conda.io/projects/conda-build/en/latest/resources/activate-scripts.html) not being compatible with their shell, and the docs include a prescient warning:

>It is generally recommended to avoid using activate scripts when another option is possible because people do not always activate environments the expected way and these packages may then misbehave.

I'm trying to understand what alternatives are available to me, and I came across some [pre-existing discussion](https://github.com/conda/conda/issues/6820) which included mention of a mechanism to set envvars using a JSON key-value mapping.

I found the [PR that introduced this functionality](https://github.com/conda/conda/pull/9157/files), but I haven't found any documentation. Is this mechanism deliberately un-documented because it's unsupported? Or is it anticipated that the recipe format will be updated to support this feature to provide a more mature API (https://github.com/conda/conda-build/issues/4833)?

After consideration of the above, I'd be happy to help document what it would look like to create this JSON mapping at build-time using a script. I think it would be especially useful to update the Activate Scripts page which I quoted above to point out preferred mechanism(s).

Thanks for your consideration!

PS: While I'm in there, I see the prose on this doc page includes "The installed scripts are recommended to be prefixed by the package name and a separating -.", but both examples use an `_` instead. Which is the true intent of the docs? I'll fix that too :)

**Comment by mfisher87** [comment]
After a bit of source-digging, I see that the [Fish shell, among many others, seems to be supported](https://github.com/conda/conda/blob/f709c9655b413902a7ae093f8e35799dd3c1d69d/conda/activate.py#L1073) for setting envvars from these JSON files, eliminating the need for activation scripts. I'm feeling really good about this option for my recipe now!

**Comment by travishathaway** [comment]
Hi @mfisher87,

Would you recommend us to update this page in any way?

- https://docs.conda.io/projects/conda-build/en/stable/resources/activate-scripts.html

**Comment by mfisher87** [comment]
@travishathaway yes, I think that's a good option (assuming this is a feature Conda _wants_ to advertise). In that case, I feel it would make the most sense to rename the page to "activation automations" or something more general. I think it should recommend:

* If you only need to set envvars, use `envvars.d` to maximize compatibility
* If you need to do arbitrary stuff, use activation scripts, but beware that they will only work for compatible shells.

And then explain the details of how to implement each.

**Comment by mfisher87** [comment]
@travishathaway I see you've marked this `good first issue`, and to me that implies that my recommendation above is sound. Can you confirm explicitly for me, and then I'll start work on this doc page? Thank you!

**Comment by travishathaway** [comment]
@mfisher87,

Apologize for my delayed response (oh how the years go by! 😂).

Yes, this would be a good issue for you to work on. Let me know if you're still interested, and I can assign it to you.

**Comment by mfisher87** [comment]
Hey Travis, thanks for your response! I understand how it is :) Please assign and I'll take a stab at the docs!
